feat(Popper): add `autoSize` prop to make it responsive in zoom/small viewport

YuanboXue-Amber opened the pull request:
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

When popups appears on top or left of the trigger, and there's not enough space in the view port, popups can be cropped.
![HNusGpaVGW](https://user-images.githubusercontent.com/28751745/109130707-56125e00-7752-11eb-8d68-07c35166705b.gif)

This PR uses [popper-max-size-modifier](https://github.com/atomiks/popper.js/tree/master/packages/popper-max-size-modifier) to give popup max-height and max-width. When `autoSize` prop is set to true, the max-height/max-weight is applied to popper. And the popper content will scroll when there's overflow.
![gQbOTv41Qv (1)](https://user-images.githubusercontent.com/28751745/109029798-7e06b080-76c3-11eb-81fa-7444e8afd675.gif)


#### Focus areas to test

(optional)

fabricteam commented:
## [Perf Analysis](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Avatar-mount_base.html">1138</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Avatar-mount.html">1125</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/BaseButton-mount_base.html">889</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/BaseButton-mount.html">910</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Breadcrumb-mount_base.html">44513</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Breadcrumb-mount.html">44623</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ButtonNext</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/ButtonNext-mount_base.html">682</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/ButtonNext-mount.html">666</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Checkbox-mount_base.html">1500</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Checkbox-mount.html">1501</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/CheckboxBase-mount_base.html">1223</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/CheckboxBase-mount.html">1254</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/ChoiceGroup-mount_base.html">4754</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/ChoiceGroup-mount.html">4748</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/ComboBox-mount_base.html">970</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/ComboBox-mount.html">953</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/CommandBar-mount_base.html">10171</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/CommandBar-mount.html">10205</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/ContextualMenu-mount_base.html">6147</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/ContextualMenu-mount.html">6206</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/DefaultButton-mount_base.html">1129</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/DefaultButton-mount.html">1119</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/DetailsRow-mount_base.html">3560</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/DetailsRow-mount.html">3614</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/DetailsRowFast-mount_base.html">3628</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/DetailsRowFast-mount.html">3599</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/DetailsRowNoStyles-mount_base.html">3375</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/DetailsRowNoStyles-mount.html">3452</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Dialog-mount_base.html">1450</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Dialog-mount.html">1464</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/DocumentCardTitle-mount_base.html">1824</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/DocumentCardTitle-mount.html">1847</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Dropdown-mount_base.html">3324</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Dropdown-mount.html">3253</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/FocusTrapZone-mount_base.html">1782</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/FocusTrapZone-mount.html">1811</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/FocusZone-mount_base.html">1793</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/FocusZone-mount.html">1810</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/IconButton-mount_base.html">1730</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/IconButton-mount.html">1729</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Label-mount_base.html">326</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Label-mount.html">342</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Layer-mount_base.html">1745</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Layer-mount.html">1763</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Link-mount_base.html">477</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Link-mount.html">457</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/MakeStyles-mount_base.html">2022</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/MakeStyles-mount.html">2022</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/MenuButton-mount_base.html">1478</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/MenuButton-mount.html">1480</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/MessageBar-mount_base.html">2022</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/MessageBar-mount.html">2041</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Nav-mount_base.html">3227</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Nav-mount.html">3277</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/OverflowSet-mount_base.html">1034</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/OverflowSet-mount.html">1053</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Panel-mount_base.html">1425</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Panel-mount.html">1438</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Persona-mount_base.html">830</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Persona-mount.html">827</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Pivot-mount_base.html">1390</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Pivot-mount.html">1371</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/PrimaryButton-mount_base.html">1292</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/PrimaryButton-mount.html">1286</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Rating-mount_base.html">7455</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Rating-mount.html">7394</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/SearchBox-mount_base.html">1343</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/SearchBox-mount.html">1295</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Shimmer-mount_base.html">2501</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Shimmer-mount.html">2536</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Slider-mount_base.html">1942</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Slider-mount.html">1988</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/SpinButton-mount_base.html">4947</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/SpinButton-mount.html">4944</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Spinner-mount_base.html">423</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Spinner-mount.html">413</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/SplitButton-mount_base.html">3152</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/SplitButton-mount.html">3171</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Stack-mount_base.html">506</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Stack-mount.html">502</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/StackWithIntrinsicChildren-mount_base.html">1538</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/StackWithIntrinsicChildren-mount.html">1574</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/StackWithTextChildren-mount_base.html">4455</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/StackWithTextChildren-mount.html">4500</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/SwatchColorPicker-mount_base.html">10166</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/SwatchColorPicker-mount.html">10154</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Tabs</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Tabs-mount_base.html">1379</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Tabs-mount.html">1413</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/TagPicker-mount_base.html">2781</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/TagPicker-mount.html">2820</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/TeachingBubble-mount_base.html">11561</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/TeachingBubble-mount.html">11559</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Text-mount_base.html">402</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Text-mount.html">432</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/TextField-mount_base.html">1382</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/TextField-mount.html">1346</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/ThemeProvider-mount_base.html">1185</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/ThemeProvider-mount.html">1178</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/ThemeProvider-virtual-rerender_base.html">599</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/ThemeProvider-virtual-rerender.html">594</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProviderNext</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/ThemeProviderNext-mount_base.html">1986</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/ThemeProviderNext-mount.html">1978</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Toggle-mount_base.html">780</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/Toggle-mount.html">793</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/buttonNative-mount_base.html">113</a></td>
            <td><a href="http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test/buttonNative-mount.html">115</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

## Perf Analysis (Fluent)
:warning: 1 potential perf regressions detected

### Potential regressions comparing to master

Scenario | Current PR Ticks | Baseline Ticks | Ratio | Regression Analysis
:--- | ---:| ---:| ---: | ---: 
AccordionMinimalPerf.default | [164](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/AccordionMinimalPerf.default.html) | [185](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/AccordionMinimalPerf.default_base.html) | 0.89:1 | [analysis](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/AccordionMinimalPerf.default.regression.txt)
<details><summary>Perf comparison</summary>

Status | Scenario | Fluent TPI | Fabric TPI | Ratio | Iterations | Ticks
:---: | :--- | ---:| ---:| ---:| ---:| ---:
🦄 | Avatar.Fluent | [0.18](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Avatar.Fluent.html) | 0.47 | 0.38:1 | 2000 | 354
🦄 | Button.Fluent | [0.12](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Button.Fluent.html) | 0.2 | 0.6:1 | 5000 | 587
🔧 | Checkbox.Fluent | [0.65](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Checkbox.Fluent.html) | 0.34 | 1.91:1 | 1000 | 651
🎯 | Dialog.Fluent | [0.16](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Dialog.Fluent.html) | 0.21 | 0.76:1 | 5000 | 809
🔧 | Dropdown.Fluent | [3.11](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Dropdown.Fluent.html) | 0.41 | 7.59:1 | 1000 | 3110
🔧 | Icon.Fluent | [0.14](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Icon.Fluent.html) | 0.06 | 2.33:1 | 5000 | 679
🦄 | Image.Fluent | [0.08](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Image.Fluent.html) | 0.13 | 0.62:1 | 5000 | 404
🔧 | Slider.Fluent | [1.61](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Slider.Fluent.html) | 0.44 | 3.66:1 | 1000 | 1610
🔧 | Text.Fluent | [0.08](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Text.Fluent.html) | 0.03 | 2.67:1 | 5000 | 381
🦄 | Tooltip.Fluent | [0.12](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Tooltip.Fluent.html) | 0.9 | 0.13:1 | 5000 | 589
>🔧 Needs work &nbsp; &nbsp; 🎯 On target &nbsp; &nbsp; 🦄 Amazing

</details>
<details><summary>Perf tests with no regressions</summary>

Scenario | Current PR Ticks | Baseline Ticks | Ratio
:--- | ---:| ---:| ---:
DatepickerMinimalPerf.default | [52466](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/DatepickerMinimalPerf.default.html) | [47404](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/DatepickerMinimalPerf.default_base.html) | 1.11:1
TreeWith60ListItems.default | [197](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/TreeWith60ListItems.default.html) | [184](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/TreeWith60ListItems.default_base.html) | 1.07:1
ChatDuplicateMessagesPerf.default | [394](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ChatDuplicateMessagesPerf.default.html) | [371](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ChatDuplicateMessagesPerf.default_base.html) | 1.06:1
HeaderMinimalPerf.default | [389](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/HeaderMinimalPerf.default.html) | [368](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/HeaderMinimalPerf.default_base.html) | 1.06:1
SkeletonMinimalPerf.default | [413](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/SkeletonMinimalPerf.default.html) | [390](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/SkeletonMinimalPerf.default_base.html) | 1.06:1
AlertMinimalPerf.default | [331](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/AlertMinimalPerf.default.html) | [315](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/AlertMinimalPerf.default_base.html) | 1.05:1
AvatarMinimalPerf.default | [221](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/AvatarMinimalPerf.default.html) | [212](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/AvatarMinimalPerf.default_base.html) | 1.04:1
ButtonMinimalPerf.default | [190](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ButtonMinimalPerf.default.html) | [182](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ButtonMinimalPerf.default_base.html) | 1.04:1
ButtonSlotsPerf.default | [610](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ButtonSlotsPerf.default.html) | [589](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ButtonSlotsPerf.default_base.html) | 1.04:1
GridMinimalPerf.default | [374](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/GridMinimalPerf.default.html) | [359](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/GridMinimalPerf.default_base.html) | 1.04:1
RadioGroupMinimalPerf.default | [470](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/RadioGroupMinimalPerf.default.html) | [453](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/RadioGroupMinimalPerf.default_base.html) | 1.04:1
Avatar.Fluent | [354](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Avatar.Fluent.html) | [339](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Avatar.Fluent_base.html) | 1.04:1
ButtonOverridesMissPerf.default | [1774](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ButtonOverridesMissPerf.default.html) | [1723](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ButtonOverridesMissPerf.default_base.html) | 1.03:1
ButtonUseCssNestingPerf.default | [1138](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ButtonUseCssNestingPerf.default.html) | [1100](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ButtonUseCssNestingPerf.default_base.html) | 1.03:1
CardMinimalPerf.default | [624](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/CardMinimalPerf.default.html) | [606](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/CardMinimalPerf.default_base.html) | 1.03:1
ChatMinimalPerf.default | [651](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ChatMinimalPerf.default.html) | [632](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ChatMinimalPerf.default_base.html) | 1.03:1
ChatWithPopoverPerf.default | [469](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ChatWithPopoverPerf.default.html) | [455](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ChatWithPopoverPerf.default_base.html) | 1.03:1
ImageMinimalPerf.default | [411](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ImageMinimalPerf.default.html) | [399](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ImageMinimalPerf.default_base.html) | 1.03:1
LayoutMinimalPerf.default | [422](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/LayoutMinimalPerf.default.html) | [409](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/LayoutMinimalPerf.default_base.html) | 1.03:1
ListCommonPerf.default | [680](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ListCommonPerf.default.html) | [662](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ListCommonPerf.default_base.html) | 1.03:1
ListNestedPerf.default | [602](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ListNestedPerf.default.html) | [583](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ListNestedPerf.default_base.html) | 1.03:1
MenuButtonMinimalPerf.default | [1622](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/MenuButtonMinimalPerf.default.html) | [1574](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/MenuButtonMinimalPerf.default_base.html) | 1.03:1
TableMinimalPerf.default | [434](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/TableMinimalPerf.default.html) | [420](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/TableMinimalPerf.default_base.html) | 1.03:1
DialogMinimalPerf.default | [816](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/DialogMinimalPerf.default.html) | [801](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/DialogMinimalPerf.default_base.html) | 1.02:1
DividerMinimalPerf.default | [397](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/DividerMinimalPerf.default.html) | [388](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/DividerMinimalPerf.default_base.html) | 1.02:1
ListMinimalPerf.default | [532](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ListMinimalPerf.default.html) | [521](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ListMinimalPerf.default_base.html) | 1.02:1
SegmentMinimalPerf.default | [386](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/SegmentMinimalPerf.default.html) | [377](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/SegmentMinimalPerf.default_base.html) | 1.02:1
VideoMinimalPerf.default | [663](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/VideoMinimalPerf.default.html) | [651](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/VideoMinimalPerf.default_base.html) | 1.02:1
Button.Fluent | [587](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Button.Fluent.html) | [578](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Button.Fluent_base.html) | 1.02:1
Checkbox.Fluent | [651](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Checkbox.Fluent.html) | [640](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Checkbox.Fluent_base.html) | 1.02:1
Tooltip.Fluent | [589](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Tooltip.Fluent.html) | [580](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Tooltip.Fluent_base.html) | 1.02:1
AttachmentSlotsPerf.default | [1230](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/AttachmentSlotsPerf.default.html) | [1220](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/AttachmentSlotsPerf.default_base.html) | 1.01:1
PortalMinimalPerf.default | [172](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/PortalMinimalPerf.default.html) | [170](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/PortalMinimalPerf.default_base.html) | 1.01:1
ProviderMinimalPerf.default | [1016](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ProviderMinimalPerf.default.html) | [1005](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ProviderMinimalPerf.default_base.html) | 1.01:1
ReactionMinimalPerf.default | [422](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ReactionMinimalPerf.default.html) | [416](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ReactionMinimalPerf.default_base.html) | 1.01:1
StatusMinimalPerf.default | [741](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/StatusMinimalPerf.default.html) | [737](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/StatusMinimalPerf.default_base.html) | 1.01:1
TableManyItemsPerf.default | [2052](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/TableManyItemsPerf.default.html) | [2025](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/TableManyItemsPerf.default_base.html) | 1.01:1
TextMinimalPerf.default | [372](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/TextMinimalPerf.default.html) | [369](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/TextMinimalPerf.default_base.html) | 1.01:1
CustomToolbarPrototype.default | [3854](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/CustomToolbarPrototype.default.html) | [3806](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/CustomToolbarPrototype.default_base.html) | 1.01:1
TooltipMinimalPerf.default | [852](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/TooltipMinimalPerf.default.html) | [840](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/TooltipMinimalPerf.default_base.html) | 1.01:1
Dialog.Fluent | [809](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Dialog.Fluent.html) | [801](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Dialog.Fluent_base.html) | 1.01:1
Icon.Fluent | [679](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Icon.Fluent.html) | [672](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Icon.Fluent_base.html) | 1.01:1
ButtonUseCssPerf.default | [832](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ButtonUseCssPerf.default.html) | [835](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ButtonUseCssPerf.default_base.html) | 1:1
CarouselMinimalPerf.default | [494](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/CarouselMinimalPerf.default.html) | [495](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/CarouselMinimalPerf.default_base.html) | 1:1
HeaderSlotsPerf.default | [807](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/HeaderSlotsPerf.default.html) | [810](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/HeaderSlotsPerf.default_base.html) | 1:1
InputMinimalPerf.default | [1308](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/InputMinimalPerf.default.html) | [1308](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/InputMinimalPerf.default_base.html) | 1:1
ItemLayoutMinimalPerf.default | [1231](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ItemLayoutMinimalPerf.default.html) | [1232](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ItemLayoutMinimalPerf.default_base.html) | 1:1
ListWith60ListItems.default | [642](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ListWith60ListItems.default.html) | [640](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ListWith60ListItems.default_base.html) | 1:1
MenuMinimalPerf.default | [907](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/MenuMinimalPerf.default.html) | [903](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/MenuMinimalPerf.default_base.html) | 1:1
PopupMinimalPerf.default | [726](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/PopupMinimalPerf.default.html) | [724](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/PopupMinimalPerf.default_base.html) | 1:1
SliderMinimalPerf.default | [1628](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/SliderMinimalPerf.default.html) | [1620](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/SliderMinimalPerf.default_base.html) | 1:1
SplitButtonMinimalPerf.default | [3827](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/SplitButtonMinimalPerf.default.html) | [3814](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/SplitButtonMinimalPerf.default_base.html) | 1:1
Dropdown.Fluent | [3110](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Dropdown.Fluent.html) | [3123](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Dropdown.Fluent_base.html) | 1:1
Image.Fluent | [404](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Image.Fluent.html) | [405](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Image.Fluent_base.html) | 1:1
AttachmentMinimalPerf.default | [175](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/AttachmentMinimalPerf.default.html) | [177](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/AttachmentMinimalPerf.default_base.html) | 0.99:1
BoxMinimalPerf.default | [380](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/BoxMinimalPerf.default.html) | [382](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/BoxMinimalPerf.default_base.html) | 0.99:1
CheckboxMinimalPerf.default | [2873](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/CheckboxMinimalPerf.default.html) | [2891](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/CheckboxMinimalPerf.default_base.html) | 0.99:1
DropdownMinimalPerf.default | [3097](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/DropdownMinimalPerf.default.html) | [3140](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/DropdownMinimalPerf.default_base.html) | 0.99:1
EmbedMinimalPerf.default | [4282](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/EmbedMinimalPerf.default.html) | [4314](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/EmbedMinimalPerf.default_base.html) | 0.99:1
LoaderMinimalPerf.default | [738](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/LoaderMinimalPerf.default.html) | [743](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/LoaderMinimalPerf.default_base.html) | 0.99:1
ProviderMergeThemesPerf.default | [1627](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ProviderMergeThemesPerf.default.html) | [1641](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ProviderMergeThemesPerf.default_base.html) | 0.99:1
IconMinimalPerf.default | [683](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/IconMinimalPerf.default.html) | [693](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/IconMinimalPerf.default_base.html) | 0.99:1
ToolbarMinimalPerf.default | [988](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ToolbarMinimalPerf.default.html) | [994](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/ToolbarMinimalPerf.default_base.html) | 0.99:1
Slider.Fluent | [1610](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Slider.Fluent.html) | [1632](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Slider.Fluent_base.html) | 0.99:1
Text.Fluent | [381](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Text.Fluent.html) | [385](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/Text.Fluent_base.html) | 0.99:1
DropdownManyItemsPerf.default | [724](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/DropdownManyItemsPerf.default.html) | [739](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/DropdownManyItemsPerf.default_base.html) | 0.98:1
FlexMinimalPerf.default | [315](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/FlexMinimalPerf.default.html) | [321](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/FlexMinimalPerf.default_base.html) | 0.98:1
LabelMinimalPerf.default | [428](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/LabelMinimalPerf.default.html) | [436](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/LabelMinimalPerf.default_base.html) | 0.98:1
FormMinimalPerf.default | [425](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/FormMinimalPerf.default.html) | [438](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/FormMinimalPerf.default_base.html) | 0.97:1
RefMinimalPerf.default | [249](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/RefMinimalPerf.default.html) | [256](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/RefMinimalPerf.default_base.html) | 0.97:1
AnimationMinimalPerf.default | [423](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/AnimationMinimalPerf.default.html) | [448](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/AnimationMinimalPerf.default_base.html) | 0.94:1
RosterPerf.default | [1130](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/RosterPerf.default.html) | [1203](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/RosterPerf.default_base.html) | 0.94:1
TextAreaMinimalPerf.default | [497](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/TextAreaMinimalPerf.default.html) | [526](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/TextAreaMinimalPerf.default_base.html) | 0.94:1
TreeMinimalPerf.default | [792](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/TreeMinimalPerf.default.html) | [840](http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/17159/merge/perf-test-northstar/TreeMinimalPerf.default_base.html) | 0.94:1

</details><div id="perfComment9423"/>

ling1726 commented:
Can this modifier also be used to make the popper cover the entire viewport ? I think there's an issue left from 0.51.x in TMP for embed that `!important` usages were needed to make that work

ling1726 commented on packages/fluentui/react-northstar/src/themes/teams/components/Popup/popupContentStyles.ts:
```diff
@@ -16,6 +16,12 @@ export const popupContentStyles: ComponentSlotStylesPrepared<PopupContentStylesP
     }),
+
+    ...(p.autoSize && {
+      // when autoSize is true, root will have inline style maxWidth and maxHeight set by popper
+      display: 'flex',
+      overflow: 'hidden',
```
Still don't understand why this is necessary sorry

YuanboXue-Amber replied:
The purpose is to make popup `content` no bigger than `root`. 

Initially I thought a `max-height` on `root` is good enough. But I found out the child(`content`) still overflows. Then I read about [this](https://stackoverflow.com/a/14263416/12667402). Shockingly when parent element has only `max-height` but not `height`, child's `max-height` computes to none. 
So I found another way to achieve the same goal, which is display flex.

YuanboXue-Amber commented:
> Can this modifier also be used to make the popper cover the entire viewport ? I think there's an issue left from 0.51.x in TMP for embed that `!important` usages were needed to make that work

I guess so. It should allow popper to be as big as it can be within the viewport. Do you know where's the issue or how can I find it? Just want to try it out in TMP to confirm

layershifter commented on packages/fluentui/docs/src/examples/components/Popup/Variations/index.tsx:
```diff
@@ -15,6 +15,11 @@ const Variations = () => (
     />
+    <ComponentExample
+      title="AutoSize"
+      description="Popup could be responsive to fit in limited space"
```
```suggestion
      description="Popup could be responsive to fit in limited space."
```

layershifter commented:
@jurokapsiar @YuanboXue-Amber is there an agreement that autosizing behavior should be enabled default? As I see it's currently enabled by default everywhere.

layershifter commented on scripts/dangerjs/detectNonApprovedDependencies/approvedPackages.ts:
```diff
@@ -84,6 +84,7 @@ export default [
   '@popperjs/core@2.1.1',
   '@popperjs/core@2.4.0',
   '@popperjs/core@2.4.2',
+  'popper-max-size-modifier@0.2.0',
```
Has friendly MIT license 🐱

YuanboXue-Amber replied:
Great that you checked it, cause I did not...

layershifter commented on packages/fluentui/docs/src/examples/components/Popup/Variations/PopupExampleAutoSize.shorthand.tsx:
```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { Button, Popup, Image, Flex, Text } from '@fluentui/react-northstar';
+import { MoreIcon } from '@fluentui/react-icons-northstar';
+
+const longContent = (
+  <Flex column gap="gap.small">
+    <Image src="https://fabricweb.azureedge.net/fabric-website/assets/images/wireframe/square-image.png" />
+    <Text content="Lorem ipsum dolor" />
+  </Flex>
+);
+
+const PopupExampleAutoSize = () => (
+  <div
+    style={{
+      paddingTop: '10rem',
+      maxWidth: '40rem',
+      border: 'red dashed',
+      position: 'relative',
+      overflow: 'hidden',
+    }}
+  >
+    <Popup
+      trigger={<Button icon={<MoreIcon />} title="Show popup" />}
+      content={longContent}
+      position={'above'}
+      inline
+      autoSize
+    />
+  </div>
+);
+
+export default PopupExampleAutoSize;
```
```suggestion
import * as React from 'react';
import { Button, Popup, Image, Flex, Text } from '@fluentui/react-northstar';
import { MoreIcon } from '@fluentui/react-icons-northstar';

const PopupExampleAutoSize = () => (
  <div
    style={{
      paddingTop: '10rem',
      maxWidth: '40rem',
      border: 'red dashed',
      position: 'relative',
      overflow: 'hidden',
    }}
  >
    <Popup
      trigger={<Button icon={<MoreIcon />} title="Show popup" />}
      content={
        <Flex column gap="gap.small">
          <Image src="https://fabricweb.azureedge.net/fabric-website/assets/images/wireframe/square-image.png" />
          <Text content="Lorem ipsum dolor" />
        </Flex>
      }
      position={'above'}
      inline
      autoSize
    />
  </div>
);

export default PopupExampleAutoSize;
```

nit: Can we inline `longContent`?

layershifter commented on packages/fluentui/react-northstar/src/utils/positioner/Popper.tsx:
```diff
@@ -292,6 +294,19 @@ export const Popper: React.FunctionComponent<PopperProps> = props => {
+          requires: ['maxSize'],
+          fn({ state }) {
+            const { height, width } = state.modifiersData.maxSize;
+            state.styles.popper.maxHeight = `${height}px`;
+            state.styles.popper.maxWidth = `${width}px`;
```
Q: _How this will be cleaned up?_
A: It works, styles are removed.

A snippet for testing:

```tsx
import * as React from "react";
import {
  Button,
  Checkbox,
  Popup,
  Image,
  Flex,
  Text
} from "@fluentui/react-northstar";
import { MoreIcon } from "@fluentui/react-icons-northstar";

const PopupExampleAutoSize = () => {
  const [autoSize, setAutoSize] = React.useState();
  const [open, setOpen] = React.useState();

  return (
    <div
      style={{
        paddingTop: "10rem",
        maxWidth: "40rem",
        border: "red dashed",
        position: "relative",
        overflow: "hidden"
      }}
    >
      <Checkbox
        label="Autosize"
        checked={autoSize}
        onChange={(e, data) => setAutoSize(data.checked)}
      />
      <Checkbox
        label="Open"
        checked={open}
        onChange={(e, data) => setOpen(data.checked)}
      />
      <Popup
        trigger={<Button icon={<MoreIcon />} title="Show popup" />}
        content={
          <Flex column gap="gap.small">
            <Image src="https://fabricweb.azureedge.net/fabric-website/assets/images/wireframe/square-image.png" />
            <Text content="Lorem ipsum dolor" />
          </Flex>
        }
        open={open}
        position={"above"}
        inline
        autoSize={autoSize}
      />
    </div>
  );
};

export default PopupExampleAutoSize;
```

layershifter replied:
Is it enough for other components that have this in their interface? For example, `ToolbarItem`, `Dropdown`, etc.

ecraig12345 reviewed (approved):
Approving lock file changes (we should add someone from Prague who can do that too)

layershifter commented on packages/fluentui/react-northstar/src/themes/teams/components/Popup/popupContentStyles.ts:
```diff
@@ -47,5 +53,9 @@ export const popupContentStyles: ComponentSlotStylesPrepared<PopupContentStylesP
       pointerEvents: 'all',
     }),
+
+    ...(p.autoSize && {
+      overflow: 'auto',
+    }),
```
```suggestion
    ...(p.autoSize && {
      overflow: 'auto',
    }),
    ...(p.autoSize === 'height' && {
    }),
    ...(p.autoSize === 'width' && {
    }),
```

Should we have separate conditions for `autoSize="height"` and `autoSize="width"` to apply overflow for a proper axis?

layershifter commented on packages/fluentui/react-northstar/src/utils/positioner/types.ts:
```diff
@@ -121,6 +121,13 @@ export interface PositioningProps {
   unstable_pinned?: boolean;
+
+  /**
+   * Change the size of the popper to fit it within the available space in viewport.
+   * true enables this for both width and height.
+   * 'height' enables this only for height and 'width' enables this only for width
```
I suggest to update there description to clarify that it applies only `maxWidth` and/or `maxHeight` as there is no guarantee that it will work in the same way for other components (`Menu`, `Toolbar`) like for `Popup` that has tweaked styles for `overflow`.

YuanboXue-Amber replied:
Nice suggestion. I changed it to be:
```ts
  /**
   * Applies max-height and max-width on popper to fit it within the available space in viewport.
   * true enables this for both width and height.
   * 'height' applies only `max-height` and 'width' for `max-width`
   */
  autoSize?: 'height' | 'width' | boolean;
```

layershifter reviewed (approved):
Approved with minor suggestions, great job, glad to see that we finally will have this behavior ❤️

YuanboXue-Amber commented:
Hi @layershifter I added some change since your last review. See this commit:
[use customized maxsize modifier, to avoid problem with 'preventOverflow'](https://github.com/microsoft/fluentui/pull/17159/commits/f5f90d942335ce24a7e4ff13ad9f8dfb114125b4)

**What I did in this commit:**
I duplicated the code in popper-maxsize-modifer. And added a step to apply max-height/max-wdith only when overflow is detected.

**Why I did this:**
I see in TMP chat, in mobile view:
- user click on chat's gif picker and popup shows. It initially renders a 2px*2px small square (I've no idea why this happens). 
- `maxsize` runs and calculates overflow using popper's `detectOverflow` function. `detectOverflow` returns `{top: number, right: number, bottom: number, left: number}`. All numbers are negative, indicating there's no overflow, because the 2*2 popup is really tiny. However `maxsize` still calculates some height and width. But because no overflow happens, the height/width calculated by maxsize is smaller than viewport. 
- Then popup gains normal size (not 2*2 anymore). But it already has the max-height and max-width applied, so it cannot extend to be bigger, even if there's space in viewport

layershifter commented on packages/fluentui/react-northstar/src/utils/positioner/Popper.tsx:
```diff
@@ -292,6 +293,35 @@ export const Popper: React.FunctionComponent<PopperProps> = props => {
+          enabled: true,
+          phase: 'beforeWrite' as PopperJs.ModifierPhases,
+          requiresIfExists: ['offset', 'preventOverflow', 'flip'],
+          options: {
+            altBoundary: true,
+            boundary: getBoundary(contentRef.current, overflowBoundary),
+          },
```
```suggestion
```

Do we need these `options`? I don't see them used in a modifier...

layershifter commented on packages/fluentui/react-northstar/src/utils/positioner/Popper.tsx:
```diff
@@ -292,6 +293,35 @@ export const Popper: React.FunctionComponent<PopperProps> = props => {
+            const { width, height } = state.rects.popper;
+            const [basePlacement] = state.placement.split('-');
+
+            const widthProp = basePlacement === 'left' ? 'left' : 'right';
+            const heightProp = basePlacement === 'top' ? 'top' : 'bottom';
```
```suggestion
            const widthProp: keyof PopperJs.SideObject = basePlacement === 'left' ? 'left' : 'right';
            const heightProp: keyof PopperJs.SideObject = basePlacement === 'top' ? 'top' : 'bottom';
```

nit, it seems that Popper defines a type for this. Can we try to reuse it?

https://github.com/popperjs/popper-core/blob/509875b0392c036739f4fe85fa580e2f265c9def/src/utils/detectOverflow.js#L36

layershifter commented on packages/fluentui/react-northstar/src/utils/positioner/Popper.tsx:
```diff
@@ -292,6 +293,35 @@ export const Popper: React.FunctionComponent<PopperProps> = props => {
+
+        autoSize && {
+          // Similar code as popper-maxsize-modifier: https://github.com/atomiks/popper.js/blob/master/src/modifiers/maxSize.js
+          // popper-maxsize-modifier only calculates the max sizes. This modifier applies the max sizes when overflow is detected
```
Just wondering, should we submit a PR with this change?

YuanboXue-Amber replied:
Sounds like a good idea. I'll make one and link this PR

layershifter commented:
> **What I did in this commit:**
> I duplicated the code in popper-maxsize-modifer. And added a step to apply max-height/max-wdith only when overflow is detected.

Makes sense to me 👍

ling1726 commented on packages/fluentui/react-northstar/src/utils/positioner/Popper.tsx:
```diff
@@ -292,6 +293,35 @@ export const Popper: React.FunctionComponent<PopperProps> = props => {
         },
         popperInitialPositionFix.modifier,
+
+        autoSize && {
```
nit: Should we put this modifier in another module for an easier read ?

ling1726 replied:
just to be clear **module** not package

ling1726 commented on scripts/dangerjs/detectNonApprovedDependencies/approvedPackages.ts:
```diff
@@ -84,6 +84,7 @@ export default [
   '@popperjs/core@2.1.1',
   '@popperjs/core@2.4.0',
   '@popperjs/core@2.4.2',
+  '@popperjs/core@2.4.3',
```
curious: not entirely sure why we keep multiple approve approved versions
